### PR TITLE
*: Fix USE_JEMALLOC macro

### LINKED ISSUE #8835: Support thread-wise memory alloc/dealloc monitor
## Enhancement

Basicly https://github.com/tikv/tikv/issues/8235, but with some tricks to fix the two copies of jemalloc problem

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #8835

Problem Summary:

### What is changed and how it works?

```commit-message
Fix USE_JEMALLOC macro
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Common/MemoryAllocTrace.cpp [hunk 1]
```diff
@@ -16,14 +16,17 @@
 #include <common/config_common.h> // Included for `USE_JEMALLOC`
 
 #ifdef USE_JEMALLOC
+#if USE_JEMALLOC == 1
```
Comment (yibin87) [inline dbms/src/Common/MemoryAllocTrace.cpp:19]: I don't get the point. Why add "#if USE_JEMALLOC == 1" works here?

File: dbms/src/Common/MemoryAllocTrace.cpp [hunk 2]
```diff
@@ -16,14 +16,17 @@
...
 #include <jemalloc/jemalloc.h>
 #endif
+#endif
```
Comment (Lloyd-Pottiger) [inline dbms/src/Common/MemoryAllocTrace.cpp:22]:
```suggestion
#if USE_JEMALLOC
#include <jemalloc/jemalloc.h>
#endif
```

File: dbms/src/Common/MemoryAllocTrace.cpp (hunk 1 above)
Comment (CalvinNeo) [reply dbms/src/Common/MemoryAllocTrace.cpp:19]: Because it could be 0

File: dbms/src/Common/MemoryAllocTrace.cpp [hunk 3]
```diff
@@ -16,14 +16,17 @@
...
 
 namespace DB
 {
 std::tuple<uint64_t *, uint64_t *> getAllocDeallocPtr()
 {
 #ifdef USE_JEMALLOC
+#if USE_JEMALLOC == 1
```
Comment (Lloyd-Pottiger) [inline dbms/src/Common/MemoryAllocTrace.cpp:29]:
```suggestion
#if USE_JEMALLOC
```

File: dbms/src/Common/MemoryAllocTrace.cpp [hunk 4]
```diff
@@ -34,5 +37,8 @@ std::tuple<uint64_t *, uint64_t *> getAllocDeallocPtr()
 #else
     return std::make_tuple(nullptr, nullptr);
 #endif
+#else
+    return std::make_tuple(nullptr, nullptr);
+#endif
```
Comment (Lloyd-Pottiger) [inline dbms/src/Common/MemoryAllocTrace.cpp:42]:
```suggestion
```

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**